Use isfinite instead of finite. Fixes build failure on Apple Silicon:

error: use of undeclared identifier 'finite'; did you mean 'isfinite'?

https://github.com/erincatto/box2d/commit/fbf51801d80fc389d43dc46524520e89043b6faf
--- src/modules/physics/box2d/Source/Common/b2Math.h.orig	2011-04-27 09:40:26.000000000 -0500
+++ src/modules/physics/box2d/Source/Common/b2Math.h	2024-05-08 01:53:53.000000000 -0500
@@ -60,7 +60,7 @@
 #ifdef _MSC_VER
 	return _finite(x) != 0;
 #else
-	return finite(x) != 0;
+	return isfinite(x);
 #endif
 }
 
